src/server/verification/processor/provider/ZoomProvider.js

```diff
@@ -169,24 +210,64 @@ class ZoomProvider implements IEnrollmentProvider {
   }
 
   async dispose(enrollmentIdentifier: string, customLogger = null): Promise<void> {
-    const { api, logger } = this
+    const { api, _handleNotExistsException, _supportsFeature } = this
+    const { defaultSearchIndexName } = api
+    // checking is API supports delete enrollment endpoint
+    const isDisposeSupported = await _supportsFeature(ZoomAPIFeature.DisposeEnrollment)
+
+    // eslint-disable-next-line require-await
+    await _handleNotExistsException(enrollmentIdentifier, customLogger, async () =>
+      api.removeEnrollmentFromIndex(enrollmentIdentifier, defaultSearchIndexName, customLogger)
+    )
+
+    // if delete enrollment supported, removig also facemap from the DB
+    if (isDisposeSupported) {
```

Comment:
this will not work if they add the delete operation in one of their next versions.
I suggest to simply call the delete end point and quietly handling the exception if it is a endpoint not found error

Reply:
actually it's not so simple because com.sun.net.httpserver doesn't have the router matching url+method like in express.
it matches only url then you have to check method manually. The FV flow on the facetec server works as the following JS/express code:

```javascript
app.use('/enrollment-3d', (req, res) => {
  let getOrDeleteResp

  if (req.method === 'get') getOrDeleteResp = processGet(req)

  if (getOrDeleteResp) {
    res.json(getOrDeleteResp)
    return
  }

  // enrollment implementation
})
```

so if delete enrollment isn't implemented, DELETE /enrollment-3d actually will start enroll, and will fail with 'already enrolled' or 'faceMap param not found' error. Not with 404

I did a research - we could use that to determine if delete supported

1. to call DELETE /enrollment-3d/some-fake-enrollment-identifier-we-are-sure-doesnt-exists
2. if we got 'no entry found' exception - that means delete is supported
3. if we getting another exception - that means  delete is not supported

I don't prefer to 'simply call and handle exception' solutions. It's a kind of the bad design. Better let's detect some feature at once and then just won't call it if we know it's not supported.

I'll remove my changes in the status endpoint in the custom server repo and will update getAPIFeatures to use /delete endpoint